controllers/idle_handlers.go

```diff
@@ -18,15 +18,19 @@ package controllers
 
 import (
 	"context"
+	"fmt"
 
 	ranv1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 func (r *ImageBasedUpgradeReconciler) handleAbort(ctx context.Context, ibu *ranv1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
-
-	// TODO actual steps
-	// If succeeds, return doNotRequeue
+	ostreeUndeployCommand := fmt.Sprintf("ostree admin undeploy rhcos_%s", ibu.Spec.SeedImageRef.Version)
+	utils.ExecuteChrootCmd(utils.Host, ostreeUndeployCommand)
```

Comment:
The execute functions should return err now. I didn't do it initially as we were using it in background only and error/progress checking was based on the progress file.

Reply:
do you mean we should change all the uses of execute functions in this PR?

Reply:
Yes. I made the execute function for the prep handler demo and didn't think through other use cases.

Reply:
updated